bc_obps/service/contact_service.py

```diff
@@ -37,6 +40,34 @@ def get_if_authorized(cls, user_guid: UUID, contact_id: int) -> Optional[Contact
             raise Exception(UNAUTHORIZED_MESSAGE)
         return contact
 
+    @classmethod
+    def get_with_places_assigned(cls, user_guid: UUID, contact_id: int) -> Optional[Contact]:
+        contact = cls.get_if_authorized(user_guid, contact_id)
+
+        places_assigned = []
+        if contact:
+            for contact_field in contact._meta.get_fields():
+                if isinstance(contact_field, (ManyToOneRel, ManyToManyRel)):
```

Comment:
There's also the ManyToMany**Field**, and that gives us stuff like contact documents. I don't think users can change that, so I don't know that it's useful to show here, but they won't be able to delete if PROTECT is on and a contact is being used somewhere. Thoughts?